### PR TITLE
Logs: Update logic to process logs dataPlane frame with labels field

### PR DESCRIPTION
**Pull request by ivanahuckova**
Building upon the discussion from https://github.com/grafana/dataplane/pull/37, where there have been updates to the logs dataPlane format (still in its proposal/early stage, subject to further changes) - specifically the renaming of `attributes` to `labels`. This PR focuses on updating the logic in Grafana. This update ensures that Grafana can correctly read data frames from the logs data plane, now including the `labels` field.

This change was implemented both in Grafana and Loki to facilitate testing of the updated logic (it's important to note that this feature in Loki is currently behind a toggle switch)

To test this logic, please follow these steps:

1. Enable the feature toggle `lokiLogsDataplane` in your `custom.ini` configuration file, setting it to `true`.
2. Confirm that log details, permalink, and table visualization are functioning as expected with the feature enabled.
3. Likewise, test the behavior with `lokiLogsDataplane` set to `false`.

This testing will help ensure that the new logic and dataPlane format changes work as intended and do not disrupt the existing functionality.

When this is merged, I'll reach out to release team and product to agree on if this can be backported to both 10.1 and 10.2

### CODE REVIEW

**Comment by gabor** [comment]
nice work 👍 . i did not run it, but the changes look good.

File: public/app/features/logs/logsFrame.ts [hunk 1]
```diff
@@ -4,18 +4,18 @@ import { parseLegacyLogsFrame } from './legacyLogsFrame';
 
 // these are like Labels, but their values can be
 // arbitrary structures, not just strings
-export type Attributes = Record<string, unknown>;
+export type LogFrameLabels = Record<string, unknown>;
 
 // the attributes-access is a little awkward, but it's necessary
-// because there are multiple,very different dataframe-represenations.
+// because there are multiple,very different dataFrame-representations.
```
Comment (svennergr) [inline public/app/features/logs/logsFrame.ts:10]: 🙏

**Review by svennergr — APPROVED** [review]
LGTM! Tested functionality and everything seems to work properly!

File: public/app/features/logs/logsFrame.ts (hunk 1 above)
Comment (matyax) [inline public/app/features/logs/logsFrame.ts:10]:
```suggestion
// because there are multiple, very different dataFrame representations.
```

**Pull request merged**